### DISCUSSION TITLE
i\I am trying to create a yaml file to dispatch a note whenever you create a discussion, but  it wont work.

### DISCUSSION THREAD

**Discussion by nikoniko78** [community/community#144982] [Programming Help]
### Select Topic Area

Question

### Body

Here is my code. The link here is a placeholder.

    name: commenter

    on:
      discussion:
        types: [created]

    jobs:
      comment:
        runs-on: ubuntu-latest

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Add comment
        run: |
          curl -X POST -H "Authorization: token ${{ secrets.Commenter }}" \
          -H "Accept: application/vnd.github.v3+json" \
          https://api.github.com/repos/${{ github.repository }}/discussions/${{ github.event.discussion.number }}/comments \
          -d '{"body":"Thanks for starting a discussion! Please check out [this link](https://example.com) for more information."}'

**Comment by nikoniko78** [comment]
Also, I plan to move this over to send a comment to pull requests.

**Comment by spenserblack** [comment]
Is the code you pasted in your discussion exactly how it is in your workflow? The first thing that stands out to me is that the `steps:` section needs to be indented more. A *job* has steps:

```yaml
jobs:
  foo:
    runs-on: ubuntu-latest
    steps:
      # ...
```

**Comment by nikoniko78** [comment]
Thank you. Yes the code is exactly as is. My secret has all the permissions required active, also. I also noticed I was getting either a 401 error or 404 error on the comments runner when I checked status

**Reply by spenserblack** [reply]
I'm actually not sure if there is a REST endpoint for discussions. I'm having difficulty finding it documented.

Last time I tried to post discussions with the API, I had to use GraphQL. Here are [the docs](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#adddiscussioncomment). You can check out [this GraphQL mutation](https://github.com/spenserblack/actions-magic-ticket/blob/f09e96d8bd1ba38b69e01d4d004b2242ff0eff36/queries/PostDiscussionComment.graphql) for an example.
